examples/carousel/js/carousel-tablist.js

```diff
@@ -0,0 +1,373 @@
+  this.tabpanelNodes = [];
+
+  this.firstTabNode = null;
+  this.lastTabNode = null;
+  this.currentTabNode = null;
+
+  this.liveRegionNode = null;
+  this.pauseButtonNode = null;
+
+  this.playLabel = 'Start automatic slide show';
+  this.pauseLabel = 'Stop automatic slide show';
+
+  this.rotate = true;
+  this.hasFocus = false;
+  this.hasHover = false;
+  this.isStopped = false;
+  this.timeInterval = 5000;
+
+  this.liveRegionNode = node.querySelector('.carousel-items');
+
+  // initialize Centering of tab controls
+
+  // initialize tabs
+
+  var nodes = node.querySelectorAll('[role="tab"]');
+
+  for (var i = 0; i < nodes.length; i++) {
+    var n = nodes[i];
+
+    this.tabNodes.push(n);
+
+    if (!this.firstTabNode) {
+      this.firstTabNode = n;
+      this.currentTabNode = n;
+    }
+
+    this.lastTabNode = n;
+
+    n.addEventListener('keydown', this.handleTabKeydown.bind(this));
+    n.addEventListener('click', this.handleTabClick.bind(this));
+    n.addEventListener('focus', this.handleTabFocus.bind(this));
+    n.addEventListener('blur', this.handleTabBlur.bind(this));
+
+    // initialize tabpanels
+
+    var tabpanelNode = document.getElementById(n.getAttribute('aria-controls'));
+
+    if (tabpanelNode) {
+      this.tabpanelNodes.push(tabpanelNode);
+
+      // support stopping rotation when any element receives focus in the tabpanel
+      tabpanelNode.addEventListener('focusin', this.handleTabpanelFocusIn.bind(this));
+      tabpanelNode.addEventListener('focusout', this.handleTabpanelFocusOut.bind(this));
+
+      var imageLink = tabpanelNode.querySelector('.carousel-image a');
+
+      if (imageLink) {
+        imageLink.addEventListener('focus', this.handleImageLinkFocus.bind(this));
+        imageLink.addEventListener('blur', this.handleImageLinkBlur.bind(this));
+      }
+
+    }
+    else {
+      this.tabpanelNodes.push(null);
+    }
+
+  }
+
+  // Pause Button
+
+  var elem = document.querySelector('.carousel-tablist .controls button.rotation');
+  if (elem) {
+    this.pauseButtonNode = elem;
+    this.pauseButtonNode.classList.add('pause');
+    this.pauseButtonNode.setAttribute('aria-label', this.pauseLabel);
+    this.pauseButtonNode.addEventListener('click', this.handlePauseButtonClick.bind(this));
+  }
+
+  this.domNode.addEventListener('mouseover', this.handleMouseOver.bind(this));
+  this.domNode.addEventListener('mouseout', this.handleMouseOut.bind(this));
+
+  // Center Tablist Controls
+
+  if (this.tablistNode.style.textAlign != 'center') {
+    this.centerTablistControls();
+    window.addEventListener('resize', this.centerTablistControls.bind(this));
+  }
+
+  // Start rotation
+  setTimeout(this.rotateSlides.bind(this), this.timeInterval);
+
+}
+
+CarouselTablist.prototype.centerTablistControls = function () {
+  width1 = this.tablistNode.getBoundingClientRect().width;
+  width2 = this.containerNode.getBoundingClientRect().width;
+  width3 = this.pauseButtonNode.getBoundingClientRect().width;
```

Comment:
all three of these should have a `var` declaration to avoid making them globals

Reply:
@smhigley 
fixed